ompi/mca/coll/basic/coll_basic_alltoall.c

```diff
@@ -92,6 +88,7 @@ mca_coll_basic_alltoall_inter(const void *sbuf, int scount,
         err = MCA_PML_CALL(irecv(prcv + (i * rcvinc), rcount, rdtype, i,
                                  MCA_COLL_BASE_TAG_ALLTOALL, comm, rreq));
         if (OMPI_SUCCESS != err) {
+            ompi_coll_base_free_reqs(req, nreqs);
```

Comment:
@bosilca Replace
    ompi_coll_base_free_reqs(req, nreqs);
by
    ompi_coll_base_free_reqs(req, i+1);
No need to free nreqs requests since only i+1 of them have been filled